api/operator/v1/vlcluster_types.go

```diff
@@ -0,0 +1,817 @@
+}
+
+func (cr *VLInsert) ProbeScheme() string {
+	return strings.ToUpper(vmv1beta1.HTTPProtoFromFlags(cr.ExtraArgs))
+}
+
+func (cr *VLInsert) ProbePort() string {
+	return cr.Port
+}
+
+func (*VLInsert) ProbeNeedLiveness() bool {
+	return true
+}
+
+// GetMetricPath returns prefixed path for metric requests
+func (cr *VLInsert) GetMetricPath() string {
+	if cr == nil {
+		return healthPath
+	}
+	return vmv1beta1.BuildPathWithPrefixFlag(cr.ExtraArgs, metricPath)
+}
+
+// ExtraArgs returns additionally configured command-line arguments
+func (cr *VLInsert) GetExtraArgs() map[string]string {
+	return cr.ExtraArgs
+}
+
+// ServiceScrape returns overrides for serviceScrape builder
+func (cr *VLInsert) GetServiceScrape() *vmv1beta1.VMServiceScrapeSpec {
+	return cr.ServiceScrapeSpec
+}
+
+// GetAdditionalService returns AdditionalServiceSpec settings
+func (cr *VLInsert) GetAdditionalService() *vmv1beta1.AdditionalServiceSpec {
+	return cr.ServiceSpec
+}
+
+// VLInsertSyslogSpec defines syslog configuration for vlinsert
+type VLInsertSyslogSpec struct {
+	// ListenPort defines syslog server listen port
+	ListenPort int32 `json:"listenPort"`
+	// StreamFields
+	// +optional
+	StreamFields []string `json:"streamFieldsT,omitempty"`
+	// +optional
+	TLSConfig *TLSServerConfig `json:"tlsConfig,omitempty"`
+}
+
+// VLSelect defines vlselect component configuration at victoria-logs cluster
+type VLSelect struct {
+	// PodMetadata configures Labels and Annotations which are propagated to the VLSelect pods.
+	PodMetadata *vmv1beta1.EmbeddedObjectMetadata `json:"podMetadata,omitempty"`
+	// LogFormat for VLSelect to be configured with.
+	// default or json
+	// +optional
+	// +kubebuilder:validation:Enum=default;json
+	LogFormat string `json:"logFormat,omitempty"`
+	// LogLevel for VLSelect to be configured with.
+	// +optional
+	// +kubebuilder:validation:Enum=INFO;WARN;ERROR;FATAL;PANIC
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// ServiceSpec that will be added to vlselect service spec
+	// +optional
+	ServiceSpec *vmv1beta1.AdditionalServiceSpec `json:"serviceSpec,omitempty"`
+	// ServiceScrapeSpec that will be added to vlselect VMServiceScrape spec
+	// +optional
+	ServiceScrapeSpec *vmv1beta1.VMServiceScrapeSpec `json:"serviceScrapeSpec,omitempty"`
+	// PodDisruptionBudget created by operator
+	// +optional
+	PodDisruptionBudget       *vmv1beta1.EmbeddedPodDisruptionBudgetSpec `json:"podDisruptionBudget,omitempty"`
+	*vmv1beta1.EmbeddedProbes `json:",inline"`
+	// Configures horizontal pod autoscaling.
+	// +optional
+	HPA *vmv1beta1.EmbeddedHPA `json:"hpa,omitempty"`
+	// RollingUpdateStrategy defines strategy for application updates
+	// Default is OnDelete, in this case operator handles update process
+	// Can be changed for RollingUpdate
+	// +optional
+	RollingUpdateStrategy appsv1.StatefulSetUpdateStrategyType `json:"rollingUpdateStrategy,omitempty"`
+	// ClaimTemplates allows adding additional VolumeClaimTemplates for StatefulSet
```

Comment:
Why is vlselect created as a StatefulSet by default? Since it doesn't support result cache, is that in the roadmap?